xdsl/traits.py

```diff
@@ -30,3 +59,12 @@ def verify(self, op: Operation) -> None:
             )
         names = ", ".join([f"'{p.name}'" for p in self.parameters])
         raise VerifyException(f"'{op.name}' expects parent op to be one of {names}")
+
+
+class IsTerminator(OpTrait):
```

Comment:
Can you add the check that this operation terminates a block?
This is the equivalent of this code: https://mlir.llvm.org/doxygen/IR_2Operation_8cpp_source.html#l01058